File: Arrays & Hashing/0409. Longest Palindrome/0409.cpp
#pragma GCC optimize("O3","Oz","unroll-loops","inline-functions","fast-math","march=native","lto","omit-frame-pointer")
class Solution {
public:
    int longestPalindrome(string s) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        std::unordered_set<char> char_set;
        int32_t res = 0;

        for (auto& c : s) {
            if (char_set.find(c) != char_set.end()) {
                char_set.erase(c);
                res += 2;
            } else {
                char_set.insert(c);
            }
        }

        if (!char_set.empty()) res++;

        return res;
    }
};

File: Arrays & Hashing/0409. Longest Palindrome/0409.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        character_set = set()
        res = 0

        for c in s:
            if c in character_set:
                character_set.remove(c)
                res += 2
            else:
                character_set.add(c)

        if character_set:
            res += 1

        return res